README.md
# Gambio Store Connector
> Store integration module for GX shops. 

The Gambio Store Connector enables the integration of the Gambio Store service into GX shops. This repository holds all  
the code required for the module, along with all the variants included in the final released archive. 

## Installation

```sh
yarn configure 
```

## Get started

You will be needing a GX shop setup in order to use this repository. The gulp workflow offers an easy way to clone and 
use shops by using the docker task. 

```sh
gulp docker --branch 4.1_develop --port 4100 --php 7.2
```

The command will need some time to complete as it will clone, setup and bootstrap the docker images for the shop 
instance. Follow the provided instructions to get the Docker containers started.

## Building archive

The archive building procedure is based on the current state of the `src` directory.

```sh
gulp archive
```

## Additional Help 

Execute the `gulp help` command to get more information on the available tasks and development workflows.

Older shop versions will not work with the default PHP v7.2, you will therefore have to choose a compatible version.

You can change the default PHP version by providing the "--php" option, when cloning a new branch with the `gulp docker`
command (e.g. "gulp docker --branch 3.3_develop --php 5.6"). 

If the docker environment is already created with a non compatible version you can still change it by deleting the 
existing server container and image with the use of the `remove.sh` script and the 
`docker rmi {image}` command. You can re-create the containers by running the `start.sh` script again. 

You will find a list of [all supported PHP images on Docker Store](https://hub.docker.com/_/php).

During the shop installation you will have to enter the database information of the Docker container setup. Those can 
always configured beforehand in the `docker-compose.yml` file, located in the root of each Docker setup. 

Bear in mind that the host is always the name of the container, e.g. `hub-connector-mysql-4.1_develop`. 
  


archive/CHANGELOG.txt
Gambio Store Connector

Das Modul ist vorgesehen für Shops von Version 3.13.x - 4.1.x

Änderungen der Version 1.0.0 zur Vorversion des Moduls:


tools/gulp/help.js
/* --------------------------------------------------------------
 gulp_doc.js 2018-11-02
 Gambio GmbH
 http://www.gambio.de
 Copyright (c) 2018 Gambio GmbH
 Released under the GNU General Public License (Version 2)
 [http://www.gnu.org/licenses/gpl-2.0.html]
 --------------------------------------------------------------
 */

'use strict';

/**
 * Gulp Help Task
 *
 * This task will output verbose information regarding the Gulp tasks.
 *
 * @param {Gulp} gulp Gulp Instance
 * @param {Object} $ Contains the automatically loaded gulp plugins.
 *
 * @return {Function} Returns the gulp task definition.
 */
module.exports = (gulp, $) => {
	return () => {
		const information = `
\n\n\n

Hub Connector - Gulp Workflow
------------------------------

The gulp workflow provided for this repository is quite similar to GX, with the only difference being the lack of
multiple task domains (e.g. general, admin, jsengine etc). Therefore you do not need to prefix the tasks.

Docker Support:

In order to make the development and testing of multiple Connector variants easier, the Gulp workflow provides a task
that will automatically setup the requested GX Git branch locally and prepare the required docker containers. The shops
will run from the local-docker accessible ip: 172.17.0.1 and the default port is "8000", although this is something you
can change by providing the port argument (e.g. "gulp docker --branch 4.1_develop --port 4100", meaning that the shop
will accessible from http://172.17.0.1:3110). Additionally, you can clone and run multiple shop branches
at the same time.

PHP Version:

Older shop versions might have problems with newer PHP versions such as v7.2. The "gulp docker" provides a parameter
that can change the default PHP version installed to an older one (e.g. "gulp docker --branch 3.3_develop --php 5.6").

Active Variant:

The Gambio Hub Connector needs to support multiple shop versions, something that can be very cumbersome during
development. While working you can pick up the preferred variant by providing a "--variant" argument followed by a
relative/absolute path to the root directory of the Connector (e.g. "gulp dev --variant 'variants/GX 3.3.1.0 - 3.3.3.0'"),
If no argument is provided the gulp task will use the "src" directory.

Target Shop:

Working with different shops is also possible with the use of the "--target" argument. While working you can pick up the preferred path so
that files are automatically synced with every change (e.g. "gulp dev --variant 'variants/GX 3.3.1.0 - 3.3.3.0' --target docker/3.3_develop/shop/src").
You can omit the full path and only provide the branch name while working with shop clones generated with the "gulp docker" command and located
in the "docker" directory (e.g. "gulp dev --variant 'variants/GX 3.3.1.0 - 3.3.3.0' --target 3.3_develop").

Gulp Tasks:

- "gulp archive": Creates a new "Gambio Hub Connector v0.0.0.zip" archive at the root directory, based on the current "variants" directory state.
- "gulp build": Builds all the assets of the active variant.
- "gulp clean": Removes all dynamically generated assets from the active variant.
- "gulp composer": Copies the required composer files to the active variant.
- "gulp dev": Compiles the assets of the active variant and starts the file watchers.
- "gulp doc": Generates the PHP documentation for the project.
- "gulp docker": Clones and prepares a new docker environment for the target shop.
- "gulp help": Outputs information on the repository.
- "gulp jshint": Runs jshint validation over the javascript files.
- "gulp scripts": Compiles the javascript assets of the active variant.
- "gulp styles": Compiles the css assets of the active variant.
- "gulp sync": Copies the content of the active variant to the target shop environment.
- "gulp test": Executes the phpunit tests of the project.
- "gulp watch": Starts the file watches for the project.

\n\n\n`.replace(/\t/g, '');

		$.util.log(information);
	};
};
